backpack/core/derivatives/rnn.py

```diff
@@ -118,31 +123,47 @@ def _jac_mat_prod(
         self, module: RNN, g_inp: Tuple[Tensor], g_out: Tuple[Tensor], mat: Tensor
     ) -> Tensor:
         self._check_parameters(module)
+        free_axis = 1
+        N_axis, T_axis = self.get_batch_and_time_axes(module)
         V: int = mat.shape[0]
-        N: int = mat.shape[2]
-        T: int = mat.shape[1]
+        N: int = mat.shape[N_axis + free_axis]
+        T: int = mat.shape[T_axis + free_axis]
         H: int = module.hidden_size
+        # use [T, N, ...] format
+        if module.batch_first:
+            output = module.output.transpose(N_axis, T_axis)
+        else:
+            output = module.output
         _jac_mat_prod: Tensor = torch.zeros(V, T, N, H, device=mat.device)
         for t in range(T):
+            mat_select_t = (slice(None),) * (T_axis + free_axis) + (t,)
             if t == 0:
                 _jac_mat_prod[:, t, ...] = einsum(
                     "nh,hi,vni->vnh",
-                    1 - module.output[t, ...] ** 2,
+                    1 - output[t, ...] ** 2,
                     module.weight_ih_l0,
-                    mat[:, t, ...],
+                    mat[mat_select_t],
                 )
             else:
                 _jac_mat_prod[:, t, ...] = einsum(
                     "nh,vnh->vnh",
-                    1 - module.output[t, ...] ** 2,
-                    einsum("hi,vni->vnh", module.weight_ih_l0, mat[:, t, ...])
+                    1 - output[t, ...] ** 2,
+                    einsum(
+                        "hi,vni->vnh",
+                        module.weight_ih_l0,
+                        mat[mat_select_t],
+                    )
```

Comment:
One more round of extraction: Extract `mat_t = mat[mat_select_t]`, then inline `mat_select_t`